rust/src/mqtt/logger.rs

```diff
@@ -194,7 +204,7 @@ fn log_mqtt(tx: &MQTTTransaction, flags: u32, js: &mut JsonBuilder) -> Result<()
                 js.set_uint("message_id", subs.message_id as u64)?;
                 js.open_array("topics")?;
                 for t in &subs.topics {
```

Comment:
Should we limit the total length of the array to be max_log_len ?

Reply:
Like I could have 2000 topics of one character...

Reply:
I assume you mean with "total length of the array" not the number of items but the sum of the characters in all topics, i.e. in your case not print the latter 1000 items in the array. It might be very unlikely to see this in paractice but I guess we're looking now to address potential crafted traffic to exhaust resources.
I think your suggestion makes sense and I'll look into it.

Reply:
> I assume you mean with "total length of the array" not the number of items but the sum of the characters in all topics

Indeed, you got the idea :-)